### ISSUE TITLE
Add support for \autocite

### ISSUE THREAD

**Issue by stakaz** [valentjn/vscode-ltex#143]
Hello, I have a problem with getting white space warnings if when using the following construct: 

```latex
Ref.~\autocite{reference} and ...
```

Of course this could be a general misbehavior with abbreviation punctuation:

```latex
.~\somecommand ...
```

This gives me the error of possible repeated white space. How can I get rid of this? I could imagine that setup a `ignoreRuleInSentence` would be possible but I was not able to do so.

**Comment by valentjn** [comment]
LT<sub>E</sub>X ignores all unknown commands by default (but not their arguments, that's the difference to [`ltex.commands.ignore`](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexcommandsignore)). So your `.~\somecommand ...` actually becomes `.~ ...`, and the tilde is replaced with another space, i.e., `.  ...` (since a tilde is a space in L<sup>A</sup>T<sub>E</sub>X), which triggers the diagnostic.

So the problem here was that `\autocite` was not supported by LT<sub>E</sub>X. This is fixed now. BibL<sup>A</sup>T<sub>E</sub>X defines a lot of citation commands, and there are still many missing for LT<sub>E</sub>X.

---

To be a bit more nitpicky:

`Ref.~\autocite{reference} and ...` shouldn't give you a diagnostic as this is replaced by `Ref.~{reference} and ...`, which ends up as `Ref. reference and ...`, which is perfectly fine. So that was probably not the original example for which you saw this. Just a wild guess, but you probably had a newline in the actual argument without commenting it out:

```latex
Ref.~\autocite{
  reference
} and ...
```

instead of

```latex
Ref.~\autocite{%
  reference%
} and ...
```

Before replacing the tilde, the first example will result in `Ref.~ reference and ...` (two spaces), while the second one will result in `Ref.~reference and ...` (one space). (BibL<sup>A</sup>T<sub>E</sub>X automatically strips the newlines, but remember to comment most newlines out to avoid spacing issues in L<sup>A</sup>T<sub>E</sub>X.)

**Comment by stakaz** [comment]
Thanks for the quick fix. However, it really was exactly the example which caused the problem, nothing with new lines. 
![image](https://user-images.githubusercontent.com/998651/96489577-e5a57e80-123f-11eb-892e-cab06d82bcd6.png)

**Comment by valentjn** [comment]
Huh, maybe my guess was too wild then. I can't reproduce this on my system, but hey, don't solve a problem that doesn't exist (anymore) :wink:

**Comment by valentjn** [comment]
Fix released in 8.0.0.
